### ISSUE TITLE
Install SDK broken on XCode 14.2.0 executor

### ISSUE THREAD

**Issue by ajamespro** [CircleCI-Public/flutter-orb#27]
## Orb version:

2.0.0

## TLDR; 

This should be the only thing that needs to change [here](https://github.com/CircleCI-Public/flutter-orb/blob/main/src/scripts/install-sdk.sh#L21):

```diff
- echo "export PATH=$INSTALL_LOCATION/flutter/bin:$PATH" >> "$BASH_ENV"
+ echo "export PATH=$INSTALL_LOCATION/flutter/bin:\$PATH" >> "$BASH_ENV"
```

## What happened:

I think I discovered a bug, but also have a possible quick solution. Although the `flutter/install_sdk` command appears to succeed, there is some flakiness in the install and some errors in `doctor`. This manifests itself with an error any time ruby is invoked. Some examples:

Running `flutter doctor`:
![image](https://user-images.githubusercontent.com/121245612/228647823-4fe93e30-845d-4a22-9bd1-f1d377971b9e.png)

Running `rbenv version`
![image](https://user-images.githubusercontent.com/121245612/228648019-f7473eb8-c2ad-4120-9cdb-7a70a66d21a7.png)

From there, cocoapods and anything gem related breaks down in all kinds of ways, but I think it will be more helpful to show the cause and solution than to exhaust all the side effect errors that occur. 

## Additional Information:

The easiest way to replicate this is with the following config file. 

<details>
  <summary>config.yml</summary>

```yml
version: 2.1

jobs:
  basic-install:
    macos:
      xcode: 14.2.0
    steps:
      - run: rbenv version
      - flutter/install_sdk
      - run: rbenv version
  minimal-broken:
    macos:
      xcode: 14.2.0
    steps:
      - run:
          name: '$PATH variable not escaped'
          command: |
            rbenv version
            echo "export PATH=/Users/distiller/development/flutter/bin:$PATH" >> "$BASH_ENV"
            rbenv version
  minimal-working:
    macos:
      xcode: 14.2.0
    steps:
      - run:
          name: 'Escaped \$PATH variable'
          command: |
            rbenv version
            echo "export PATH=/Users/distiller/development/flutter/bin:\$PATH" >> "$BASH_ENV"
            rbenv version
orbs:
  flutter: circleci/flutter@2.0.0

workflows:
  bug-report:
    jobs:
      - basic-install
      - minimal-broken
      - minimal-working
```
</details>

![image](https://user-images.githubusercontent.com/121245612/228654656-93ae905f-6f89-404f-8e4f-157e8eb2c085.png)

In here, the job `basic-install` is an integrated example which uses the orb and shows how the install breaks the bash command: `rbenv version`. The two jobs `minimal-working` and `minimal-broken` show where the root cause of this might be occurring. I suspect that ruby is doing something with the `PATH` variable, and the flutter sdk is overwriting this variable, rather than constructing it in `$BASH_ENV`.

### Basic Install

Using the following steps, the first `rbenv version` succeeds but the second call fails just after installing:

```yml
steps:
  - run: rbenv version
  - flutter/install_sdk
  - run: rbenv version
```

![image](https://user-images.githubusercontent.com/121245612/228649895-22823014-351b-4b9e-a441-fe5be73bc043.png)

### Broken

`minimal-broken`, performing the PATH overwrite with [current code](https://github.com/CircleCI-Public/flutter-orb/blob/fa0ac2b381b5eb275fe5279ad7b2ae2de380b058/src/scripts/install-sdk.sh#L21).

```yml
command: |
  rbenv version
  echo "export PATH=/Users/distiller/development/flutter/bin:$PATH" >> "$BASH_ENV"
  rbenv version
```

![image](https://user-images.githubusercontent.com/121245612/228655989-73805dde-584d-4c48-ab68-904cb3049490.png)

### Working

vs `minimal-working`, which adds an escaped character next to the $ sign in `$PATH`

```yml
command: |
  rbenv version
  echo "export PATH=/Users/distiller/development/flutter/bin:\$PATH" >> "$BASH_ENV"
  rbenv version
```

![image](https://user-images.githubusercontent.com/121245612/228656076-9403d360-a538-4551-afeb-975afd312cda.png)

## Expected behavior:

`rbenv version` should not exit with an error after installing flutter sdk on xcode 14.2 executor.

**Comment by ryanbourdais** [comment]
Hi @ajamespro thank you so much for all the investigation you did into this issue. I have just drafted up a PR to fix this and this should be fixed in the next release.
